tests/handler_async_test.py

```diff
@@ -142,6 +145,31 @@ async def test_get_batch_details(batch_size):
     await handler.deinit()
 
 
+def _check_iterative_batch_details(ip, details, token):
+    """Helper for iterative batch tests."""
+    assert ip == details.get("ip")
+    assert "country" in details
+    assert "city" in details
+    if token:
+        assert "asn" in details or "anycast" in details
+        assert "company" in details or "org" in details
+        assert "privacy" in details or "anycast" in details
+        assert "abuse" in details or "anycast" in details
+        assert "domains" in details or "anycast" in details
+
+
+@pytest.mark.parametrize("batch_size", [None, 1, 2, 3])
+@pytest.mark.asyncio
+async def test_get_iterative_batch_details(batch_size):
+    handler, token, ips = _prepare_batch_test()
+    async for ips, details in handler.getIterativeBatchDetails(
+        ips, batch_size
+    ):
+        # results.append((ip_address, details))
+        _check_iterative_batch_details(ips, details, token)
+        await handler.deinit()
```

Comment:
How did this test case pass? This is deinitializing a handler that will get used again on the next for-loop iteration.

Reply:
Updated.